src/internal/connector/exchange/api/mail.go

```diff
@@ -107,6 +108,26 @@ func (c Mail) GetItem(
 		return nil, nil, err
 	}
 
+	if *mail.GetHasAttachments() || HasAttachments(mail.GetBody()) {
+		for count := 0; count < numberOfRetries; count++ {
+			attached, err := c.largeItem.
+				Client().
+				UsersById(user).
+				MessagesById(itemID).
+				Attachments().
+				Get(ctx, nil)
+			if err == nil {
+				mail.SetAttachments(attached.GetValue())
+				break
+			}
+		}
+
```

Comment:
Not a blocker - but for supportability - we should accumulate the errors and return those if we run out of retries.

Reply:
That feels a little bit like overpopulation of errors to me.  For example, if we had automatic retry/backoff in the client, instead of here in the code, we wouldn't add an error per retry, would we?

A log per retry, perhaps, sounds a bit more appropriate.

Reply:
A debug log per retry - would be useful - yes. This will help to debug when the issue is repro'able and we also have the option of asking the bug reporter to reproduce.

For the error message - what I was suggesting was that *if* we do end up failing the request eventually (after retries) - it might be interesting to see the entire chain of errors (from each retry) and not just the last one. This helps debug issues where it's not the same error each time but a chain of different errors. 

Not a blocker here because for the most part the last error will be the most relevant.

Reply:
Sounds good, thanks for the insight.